src/Moryx.Runtime.Kernel/FileSystem/LocalFileSystem.cs

```diff
@@ -0,0 +1,278 @@
+﻿using Castle.MicroKernel.Registration;
+using Microsoft.Extensions.Logging;
+using Moryx.FileSystem;
+using Moryx.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Moryx.Runtime.Kernel.FileSystem
+{
+    internal class LocalFileSystem : IMoryxFileSystem
+    {
+        private string _fsDirectory;
+        private string _ownerFilesDirectory;
+        private readonly ILogger _logger;
+
+        private readonly List<MoryxFileTree> _ownerTrees = new List<MoryxFileTree>();
+
+        public LocalFileSystem(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger(nameof(LocalFileSystem));
+        }
+
+        public void SetBasePath(string basePath = "fs")
+        {
+            _fsDirectory = Path.Combine(Directory.GetCurrentDirectory(), basePath);
+            _ownerFilesDirectory = Path.Combine(_fsDirectory, "owners");
+        }
+
+        public void LoadTrees()
+        {
+            // Load all trees from the owner directory
+            var ownerFiles = Directory.EnumerateFiles(_ownerFilesDirectory);
+            foreach (var ownerFile in ownerFiles)
+            {
+                var treeHash = File.ReadAllText(ownerFile);
+                var ownerTree = new MoryxFileTree
+                {
+                    FileName = ownerFile,
+                    Hash = treeHash,
+                };
+                ReadExtensibleTree(ownerTree);
+            }
+        }
+
+        public MoryxFile GetFile(string hash)
+        {
+            return _ownerTrees.FindFile(hash);
+        }
+
+        public async Task<string> WriteAsync(MoryxFile file, Stream content)
+        {
+            if (file is MoryxFileTree fileTree)
+                return await WriteTreeAsync(fileTree);
+            else if (content != null)
+                return await WriteBlobAsync(file, content);
+
+            throw new ArgumentException("For all files except trees the content stream must be given");
+        }
+
+        private async Task<string> WriteTreeAsync(MoryxFileTree tree)
+        {
+            // Convert metadata to lines 
+            var stream = new MemoryStream();
+            using (var sw = new StreamWriter(stream))
+            {
+                foreach (var line in tree.Files.Select(FileToLine))
+                    sw.WriteLine(line);
+                await sw.FlushAsync();
+            }
+
+            tree.Hash = await StreamToDiskAsync(stream);
+
+            // Update parent or owner file
+            if (tree.ParentTree == null)
```

Comment:
Did we conclude the discussion what happens if within a single owner tree multiple files are added/trees updates? 🤔